src/main/java/com/odysseusinc/arachne/portal/api/v1/controller/BaseDataNodeController.java

```diff
@@ -114,9 +119,22 @@ public BaseDataNodeController(BaseAnalysisService analysisService,
     public CommonDataNodeCreationResponseDTO createManualDataNode(
             @RequestBody @Valid CommonDataNodeRegisterDTO commonDataNodeRegisterDTO,
             Principal principal
-    ) throws PermissionDeniedException, AlreadyExistException {
+    ) throws PermissionDeniedException, AlreadyExistException, ValidationException {
 
         final DN dataNode = conversionService.convert(commonDataNodeRegisterDTO, getDataNodeDNClass());
+        final Long dataNodeId = dataNode.getId();
+        if (dataNodeId != null) {
+            return conversionService.convert(baseDataNodeService.getById(dataNodeId), CommonDataNodeCreationResponseDTO.class);
+        }
+        final Organization organization = conversionService.convert(commonDataNodeRegisterDTO.getOrganization(), Organization.class);
+        final Organization savedOrganization;
+        final Long organizationId = organization.getId();
+        if (organizationId != null) {
+            savedOrganization = organizationService.get(organizationId);
+        } else {
+            savedOrganization = organizationService.create(organization);
+        }
```

Comment:
I would create a service method like "getOrCreate"

Reply:
in the OrganizationService?

Reply:
Yes, I thought of such options